plugins/outputs/opensearch/README.md

```diff
@@ -0,0 +1,364 @@
+      }
+    }
+  }
+}
+
+```
+
+[2]: https://opensearch.org/docs/latest/opensearch/index-templates/
+
+### Example events
+
+This plugin will format the events in the following way:
+
+```json
+{
+  "@timestamp": "2017-01-01T00:00:00+00:00",
+  "measurement_name": "cpu",
+  "cpu": {
+    "usage_guest": 0,
+    "usage_guest_nice": 0,
+    "usage_idle": 71.85413456197966,
+    "usage_iowait": 0.256805341656516,
+    "usage_irq": 0,
+    "usage_nice": 0,
+    "usage_softirq": 0.2054442732579466,
+    "usage_steal": 0,
+    "usage_system": 15.04879301548127,
+    "usage_user": 12.634822807288275
+  },
+  "tag": {
+    "cpu": "cpu-total",
+    "host": "opensearhhost",
+    "dc": "datacenter1"
+  }
+}
+```
+
+```json
+{
+  "@timestamp": "2017-01-01T00:00:00+00:00",
+  "measurement_name": "system",
+  "system": {
+    "load1": 0.78,
+    "load15": 0.8,
+    "load5": 0.8,
+    "n_cpus": 2,
+    "n_users": 2
+  },
+  "tag": {
+    "host": "opensearhhost",
+    "dc": "datacenter1"
+  }
+}
+```
+
+## Configuration
+
+```toml @sample.conf
+# Configuration for Opensearch to send metrics to.
+[[outputs.opensearch]]
+  ## The full HTTP endpoint URL for your Opensearch instance
+  ## Multiple urls can be specified as part of the same cluster,
+  ## this means that only ONE of the urls will be written to each interval
+  urls = [ "http://node1.os.example.com:9200" ] # required.
+  ## Opensearch client timeout, defaults to "5s" if not set.
+  timeout = "5s"
+  ## Set to true to ask Opensearch a list of all cluster nodes,
+  ## thus it is not necessary to list all nodes in the urls config option
+  enable_sniffer = false
+  ## Set to true to enable gzip compression
+  enable_gzip = false
+  ## Set the interval to check if the Opensearch nodes are available
+  ## Setting to "0s" will disable the health check (not recommended in production)
+  health_check_interval = "10s"
+  ## Set the timeout for periodic health checks.
+  # health_check_timeout = "1s"
+  ## HTTP basic authentication details.
+  ## HTTP basic authentication details
+  # username = "telegraf"
+  # password = "mypassword"
+  ## HTTP bearer token authentication details
+  # auth_bearer_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9"
+
+  ## Index Config
+  ## The target index for metrics (Opensearch will create if it not exists).
+  ## You can use the date specifiers below to create indexes per time frame.
+  ## The metric timestamp will be used to decide the destination index name
+  # %Y - year (2016)
+  # %y - last two digits of year (00..99)
+  # %m - month (01..12)
+  # %d - day of month (e.g., 01)
+  # %H - hour (00..23)
+  # %V - week of the year (ISO week) (01..53)
+  ## Additionally, you can specify a tag name using the notation {{tag_name}}
+  ## which will be used as part of the index name. If the tag does not exist,
+  ## the default tag value will be used.
+  # index_name = "telegraf-{{host}}-%Y.%m.%d"
+  # default_tag_value = "none"
+  index_name = "telegraf-%Y.%m.%d" # required.
```

Comment:
I don't think inventing yet another templating language, which even mixes different notations will ease the use of it.

As other plugins already use the golang template mechanism, I would use this as well. Advantage would be to have more options for date specification.

See the [template processor](https://github.com/influxdata/telegraf/tree/master/plugins/processors/template) to use it's `TemplateMetric` And the example `index_name` formats will then look like this:
```toml
  index_name = 'telegraf-{{.Tag "host"}}-{{.Time.Format "2006-01-02"}}'
```

Reply:
Not a new invention, this is present in elasticsearch and supported already.

Reply:
Can you show me the docs of ElasticSearch or OpenSearch where this format is explained?

Reply:
agree, will make the change

Reply:
Thanks!

Reply:
Have made the changes, please take a look.

Reply:
@mannukalra I fully agree with @Hipska, you really should reuse what is there as this will keep the code cleaner and will free you from extension requests to "your language".

Reply:
Since @mannukalra started using golang templates, this conversation can be marked as resolved for me. (can't do it myself)